components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +327,79 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  private showInputs(inputName: string): void {
+
+    const formValues = this.expressionForm.value;
+    const matchAllWildCard = '*';
+
+    this.setFormLabels(formValues.type);
+
+    switch (inputName) {
+      case 'type':
+        this.resetFormControls();
+        if (formValues.type === 'user' || formValues.type === 'team') {
+          this.addIdentityControl();
+        } else if (formValues.type === 'token') {
+          this.addNameControl();
+        }
+        break;
+      case 'identity':
+        if (formValues.identity !== matchAllWildCard) {
+          this.addNameControl();
+        } else {
+          this.expressionForm.removeControl('name');
+        }
+        break;
+      default:
+        break;
+    }
+
+  }
```

Comment:
MAJOR overhaul in showInputs().  I'm now following Angular Core team suggestion and adding/removing form controls as we go.  This has provided a much simpler way of making sure the inputs are showing and the correct validators are applied.  Although small, I broke reused patterns out into their own functions to make showInputs() switch statement easier to read and follow.  showInputs() is now only firing on form input changes.